### PR TITLE
Developer UI: Welcome page

### PR DESCRIPTION
**Pull request by mikheevm**
Review after: 
- https://github.com/gnosis/safe-apps-sdk/pull/134

This PR:
- Adds welcome page

### CODE REVIEW

File: packages/safe-apps-developer-ui/src/pages/Welcome.tsx [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+import Button from '@material-ui/core/Button';
+import { makeStyles } from '@material-ui/core/styles';
+import { lg, md } from 'src/styles/variables';
+import { Card } from 'src/components/Card';
+import { Dot } from 'src/components/Dot';
+import { useProviderStore } from 'src/stores/provider';
+import { useContractsStore } from 'src/stores/contracts';
+import { ConnectButton } from 'src/components/ConnectButton';
+import { deployFallbackHandler, deployProxyFactory, deployMasterCopy } from 'src/api/safeContracts';
+import { getSafeContract, getProxyFactoryContract } from 'src/api/safeContracts';
```
Comment (germartinez) [inline packages/safe-apps-developer-ui/src/pages/Welcome.tsx:14]: Move imports from line 14 to line 13 or viceversa :)

File: packages/safe-apps-developer-ui/src/stores/provider.ts [hunk 2]
```diff
@@ -1,29 +1,53 @@
...
+      return set({
+        loaded: false,
+        account: '',
+        networkId: ETHEREUM_NETWORK_TO_ID.UNKNOWN,
+        provider: null,
+        signer: null,
+      });
+    }
+
+    return set({ account, loaded: true, networkId });
```
Comment (germartinez) [inline packages/safe-apps-developer-ui/src/stores/provider.ts:50]: Should the `provider` and `signer` be also added here?

File: packages/safe-apps-developer-ui/src/components/ConnectButton.tsx [hunk 3]
```diff
@@ -1,30 +1,55 @@
...
+      disabled={disabled}
+      className={className}
+    >
       Connect
     </SButton>
   );
 };
 
-export default ConnectButton;
+export { ConnectButton };
```
[the remark this answers is not in the document]
Comment (mikheevm) [reply packages/safe-apps-developer-ui/src/components/ConnectButton.tsx:55]: https://github.com/basarat/typescript-book/blob/master/docs/tips/defaultIsBad.md

just decided to give it a try

File: packages/safe-apps-developer-ui/src/App.test.tsx [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { render, screen } from '@testing-library/react';
+import App from './App';
+
+test('renders learn react link', () => {
```
[the remark this answers is not in the document]
Comment (mikheevm) [reply packages/safe-apps-developer-ui/src/App.test.tsx]: It's removed

File: packages/safe-apps-developer-ui/src/stores/provider.ts (hunk 2 above)
Comment (mikheevm) [reply packages/safe-apps-developer-ui/src/stores/provider.ts:50]: It will partially update the state, we don't need to update them

Comment (germartinez) [reply packages/safe-apps-developer-ui/src/stores/provider.ts:50]: What if the provider is updated twice?
- The first update has no account and the provider is initialized with all the properties to their default values (null, false, etc.)
- The second update has an account and the properties `account`, `loaded` and `networkId` are updated but `provider` and `signer` are still null.

As far as I understand.

Comment (mikheevm) [reply packages/safe-apps-developer-ui/src/stores/provider.ts:50]: It's only invoked when there's an existing provider and account/network changes: https://github.com/gnosis/safe-apps-sdk/pull/135/files#diff-96c3a90c6ed644ae146525eedc19cc94ac0471e7674b81c28cb24d34b4b46cf6R29-R30

**Review by germartinez — APPROVED** [review]

**Pull request merged**